### PR TITLE
Global fixes: typo, bugs and description

### PR DESCRIPTION
**Pull request by WinterSilence**

### CODE REVIEW

File: classes/Kohana/Minion/CLI.php [hunk 1]
```diff
@@ -1,10 +1,19 @@
...
+abstract class Kohana_Minion_CLI {
 
-class Kohana_Minion_CLI {
-
-	public static $wait_msg = 'Press any key to continue...';
-
-	protected static $foreground_colors = array(
+	/**
+	* @var array colour designations of the text
+	*/
```
Comment (kemo) [inline classes/Kohana/Minion/CLI.php:15]: Missing spaces before \* on 2nd and 3rd line of the docblock, please fix (same below)

File: classes/Kohana/Minion/Exception.php [hunk 2]
```diff
@@ -46,19 +36,22 @@ public static function handler(Exception $e)
 		}
 		catch (Exception $e)
 		{
-			// Clean the output buffer if one exists
-			ob_get_level() and ob_clean();
-
 			// Display the exception text
-			echo Kohana_Exception::text($e), "\n";
+			fwrite(STDERR, parent::text($e));
```
Comment (kemo) [inline classes/Kohana/Minion/Exception.php:40]: Do we need PHP_EOL here?

File: classes/Kohana/Minion/Task.php [hunk 3]
```diff
@@ -45,30 +77,38 @@ public static function convert_class_to_task($class)
...
-		if (($task = Arr::get($options, 'task')) !== NULL)
-		{
-			unset($options['task']);
-		}
-		else if (($task = Arr::get($options, 0)) !== NULL)
+		if (isset($options['task']) OR isset($options[0]))
 		{
 			// The first positional argument (aka 0) may be the task name
-			unset($options[0]);
+			$task = isset($options['task']) ? $options['task'] : $options[0];
```
Comment (kemo) [inline classes/Kohana/Minion/Task.php:105]: Arr::get($options, 'task', $options[0]);

File: guide/minion/tasks.md [hunk 4]
```diff
@@ -1,48 +1,58 @@
 # Writing Tasks
 
-Writing a task in minion is very easy. Simply create a new class called `Task_<Taskname>` and put it inside `classes/task/<taskname>.php`.
+Writing a minion tasks is very easy. 
+Simply create a new class extended [Minion_Task] and called `Task_<Taskname>` in `classes/Task/` directory.
```
Comment (kemo) [inline guide/minion/tasks.md:4]: extending\* call*

File: classes/Kohana/Minion/Task.php [hunk 5]
```diff
@@ -191,31 +198,50 @@ public function get_accepted_options()
...
 	{
 		// Add a rule to each key making sure it's in the task
 		foreach ($validation->data() as $key => $value)
 		{
-			$validation->rule($key, array($this, 'valid_option'), array(':validation', ':field'));
+			$validation->rule(
+				$key, 
+				array($this, 'valid_option'), 
+				array(':validation', ':field')
+			);
```
Comment (WinterSilence) [inline classes/Kohana/Minion/Task.php:237]: cut long string (more 120 symbols)

File: classes/Kohana/Minion/CLI.php (hunk 1 above)
Comment (WinterSilence) [inline classes/Kohana/Minion/CLI.php:15]: wtf?

File: classes/Kohana/Minion/Exception.php (hunk 2 above)
Comment (WinterSilence) [inline classes/Kohana/Minion/Exception.php:40]: for what?

File: classes/Kohana/Minion/Task.php (hunk 3 above)
Comment (WinterSilence) [inline classes/Kohana/Minion/Task.php:105]: I try not use other classes without much need to loosen the ties between them.

File: classes/Kohana/Minion/CLI.php (hunk 1 above)
Comment (kemo) [inline classes/Kohana/Minion/CLI.php:15]:
``` php
/**
 * 
 */
```

File: classes/Kohana/Minion/Exception.php (hunk 2 above)
Comment (kemo) [inline classes/Kohana/Minion/Exception.php:40]: There was a newline output on the line you fixed

File: classes/Kohana/Minion/Task.php (hunk 3 above)
Comment (kemo) [inline classes/Kohana/Minion/Task.php:105]: You used `Arr::get()` on line 340 of this same file already and Minion isn't becoming a self-contained module in this release anyway :)

**Comment by kemo** [comment]
I will merge this as soon as these few small finesses are fixed. Good job!

File: classes/Kohana/Minion/Task.php (hunk 3 above)
Comment (WinterSilence) [inline classes/Kohana/Minion/Task.php:105]: agree. fix today

**Comment by WinterSilence** [comment]
Houston, we have a problem!
I sent requests from another computer. how to change these requests?

**Comment by kemo** [comment]
@WinterSilence which requests, what seems to be the problem?

**Comment by WinterSilence** [comment]
my local version does not include these changes. how to add them?

**Comment by WinterSilence** [comment]
understood, i'm stupid :)

**Comment by kemo** [comment]
I'm not sure you can do that since you removed your github repo. I'll merge this as-is and you can pull from `kohana/3.4/develop`, fix the remaining things and send another pull request for the fixes.

Nobody is stupid here :)

**Comment by WinterSilence** [comment]
@cyrgit https://github.com/kohana/minion/blob/3.4/develop/guide/minion/tasks.md

**Comment by acoulton** [comment]
@WinterSilence that documentation you linked to is inconsistent. The code sample shows `$this->options` but the text immediately below still refers to taking a `$params` array.

To be honest, I'm with @cyrgit on this, I don't see any good reason for this breaking change, which is the second I'm aware of (removing `::password` being the other) that was introduced in a "typos and bugs" pull request. These really shouldn't have been merged without being split into separate commits with proper messages explaining what was changed, why and the implications.

**Comment by cyrrill** [comment]
API breaking changes should not be allowed when there is no good reason for it. This just seems like a unilateral decision that was pushed in with the general refactor. Why should I go and update all my Tasks  to accommodate for this? Minion was running fine before it.
Refactors are good for cleaning up the code, typos, and creating efficiency. If the API is to be changed, it should be a consensus decision, and be made a separate commit from a cleaning refactor.

**Comment by WinterSilence** [comment]
@acoulton @cyrgit if we return attribute `$params` is only confuse users because values contained in `$this->_options`. I will correct the code, if accepted by a few developers.

New version of Kohana will be very different from the previous, so full compatibility is not guaranteed. We must make a choice between backward compatibility and new user-friendly codes.

**Comment by cyrrill** [comment]
@WinterSilence  I don't think it will confuse users to have the $params array in _execute(), as that is the way it has always run. 
Confusing is to discover your several dozens of Tasks don't run anymore due to a breaking change hidden in a "typos" refactor.

**Comment by cyrrill** [comment]
On a separate issue from this PR, which shouldn't mix feature requests or breaking changes:

I think the best design for this would actually be to keep a separation between defaults and runtime parameters. Such that $this->options never changes, and whatever is passed in CLI gets sent in through $params. 

This allows _execute() to distinguish if the Task is being run with default parameters, or ones sent by the user. When set_options() runs, it overwrites the original defaults the class had defined, making it impossible to know what the default was.

**Comment by WinterSilence** [comment]
https://github.com/kohana/minion/pull/103

**Comment by cyrrill** [comment]
@WinterSilence  There is already a PR for this pending, no need to duplicate

**Pull request merged**